### PR TITLE
Extend torch->onnx export for quantized convolutional ops

### LINKED ISSUE #102747: Many missing convolutional ops in PyTorch -> ONNX exporter (e.g. quantized::conv1d, quantized::conv_transpose2d)
### 🚀 The feature, motivation and pitch

Exporting a quantized model to ONNX has several missing convolutional ops. Adding these will benefit not just immediate ONNX deploymnet, but since many use ONNX as an portable ML representation, it can benefit a variety of downstream compilation workflows.

```
import torch    
import torch.onnx    
import torch.nn as nn    
import torch.nn.functional as F    
    
DIM, TRANSPOSE, FUSE = 1, False, False   # quantized::conv1d                  (fails to export)    
#DIM, TRANSPOSE, FUSE = 1, False, True   # quantized::conv1d_relu             (supported)    
#DIM, TRANSPOSE, FUSE = 2, False, False  # quantized::conv2d                  (supported)    
#DIM, TRANSPOSE, FUSE = 2, False, True   # quantized::conv2d_relu             (supported)    
#DIM, TRANSPOSE, FUSE = 3, False, False  # quantized::conv3d                  (fails to export)    
#DIM, TRANSPOSE, FUSE = 3, False, True   # quantized::conv3d_relu             (fails to export)    
#DIM, TRANSPOSE, FUSE = 1, True, False   # quantized::conv_transpose1d        (fails to export)    
#DIM, TRANSPOSE, FUSE = 1, True, True    # quantized::conv_transpose1d_relu   (no fuser)    
#DIM, TRANSPOSE, FUSE = 2, True, False   # quantized::conv_transpose2d        (fails to export)    
#DIM, TRANSPOSE, FUSE = 2, True, True    # quantized::conv_transpose2d_relu   (no fuser)    
#DIM, TRANSPOSE, FUSE = 3, True, False   # quantized::conv_transpose3d        (fails to export)    
#DIM, TRANSPOSE, FUSE = 3, True, True    # quantized::conv_transpose3d_relu   (no fuser)    
    
    
if TRANSPOSE:    
    CONV_OP = getattr(nn, f"ConvTranspose{DIM}d")    
else:    
    CONV_OP = getattr(nn, f"Conv{DIM}d")    
    
class Model(nn.Module):    
    def __init__(self):    
        super().__init__()    
        self.conv = CONV_OP(3, 3, 3)    
        self.relu = nn.ReLU()    
        self.quant = torch.ao.quantization.QuantStub()    
        self.dequant = torch.ao.quantization.DeQuantStub()    
    
    def forward(self, x):    
        z = x    
        z = self.quant(z)    
        z = self.conv(z)    
        z = self.relu(z)    
        z = self.dequant(z)    
        return z    
    
    
model = Model()    
model.eval()    
    
input_shape = (1, 3) + (32,) * DIM    
    
model_fp32 = model    
model_fp32.qconfig = torch.ao.quantization.get_default_qconfig("qnnpack")    
if FUSE:    
    model_fp32_fused = torch.ao.quantization.fuse_modules(model_fp32, [["conv", "relu"]])    
else:    
    model_fp32_fused = model_fp32    
model_fp32_prepared = torch.ao.quantization.prepare(model_fp32_fused)    
input_fp32 = torch.randn(*input_shape)    
# PTQ    
model_fp32_prepared(input_fp32)    
    
model_int8 = torch.ao.quantization.convert(model_fp32_prepared)    
model = model_int8    
    
torch.onnx.export(    
    model,    
    input_fp32,    
    "output.onnx",    
    opset_version=16,       
    input_names=["x"],    
    output_names=["y"],    
) 
```
Only three of these can currently be exported, while another six have support in PyTorch but fails in the ONNX exporter. Three of them fail earlier in that there is not fuser for transposed convolutions and relu.

I propose to add the missing 6 and I do have a straightforward PR in the pipeline already.

### Alternatives

There are other ways to get to a quantized ONNX model, for instance from Tensorflow (I would strongly prefer Torch).

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by gustavla**
- Extend support:
  - quantized::conv1d
  - quantized::conv3d
  - quantized::conv3d_relu
  - quantized::conv_transpose1d
  - quantized::conv_transpose2d
  - quantized::conv_transpose3d
  - Note: quantized::{conv1d_relu,conv2d,conv2d_relu} already supported.
- To support this, quantization unpacking added for:
  - conv1d
  - conv_transpose1d
  - conv_transpose2d
  - conv_transpose3d
  - Note: conv3d/conv3d_relu already had weights unpacking set up, even though it didn't have torch.onnx support.
- Add tests.
- The 3D tests will fail if run with the qnnpack backend (e.g., on Apple silicon Mac), so added decorator skipIfQuantizationBackendQNNPack.
- Minor fix in `aten/src/ATen/native/quantized/cpu/qconv.cpp` for 3D convolutions (triggered by added tests).

Fixes #102747

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: aten/src/ATen/native/quantized/cpu/qconv.cpp [hunk 1]
```diff
@@ -931,7 +931,7 @@ at::Tensor PackedConvWeightsQnnp<kSpatialDim>::apply_impl(
     output_shape = MakeDeConvOutputShape<kSpatialDim>(
         N,
         M,
-        {H, W},
+        kSpatialDim == 2 ? std::vector<int64_t>{H, W} : std::vector<int64_t>{D, H, W},
```
Comment (BowenBao) [inline aten/src/ATen/native/quantized/cpu/qconv.cpp:934]: @kit1980 do you know the best person to review this part of change?

**Review by BowenBao — APPROVED** [review]
ONNX side LGTM. Need signal from @kit1980 / @malfet on `aten/src/ATen/native/quantized/cpu/qconv.cpp` change.

**Review by thiagocrepaldi — APPROVED** [review]
+1 on "ONNX side LGTM. Need signal from @kit1980 / @malfet on aten/src/ATen/native/quantized/cpu/qconv.cpp change."

**Review by kit1980 — APPROVED** [review]

**Pull request closed without merging**